tests/unit-tests/data_utils.py

```diff
@@ -115,6 +115,95 @@ def generate_dummy_hetero_graph(size='tiny', gen_mask=True):
+        ("n0", "r1", "n1"): (th.randint(data_size, (2 * data_size,)),
+                             th.randint(data_size, (2 * data_size,)))
+    }
+
+    hetero_graph = dgl.heterograph(edges, num_nodes_dict=num_nodes_dict)
+
+    # set node and edge features
+    node_feat = {'n0': th.randn(data_size, 2),
+                 'n1': th.randn(data_size, 2)}
+
+    edge_feat = {'r0': th.randn(data_size, 2),
+                 'r1': th.randn(2 * data_size, 2)}
+
+    hetero_graph.nodes['n0'].data['feat'] = node_feat['n0']
+    hetero_graph.nodes['n1'].data['feat'] = node_feat['n1']
+    hetero_graph.nodes['n0'].data['label'] = th.randint(10, (hetero_graph.number_of_nodes('n0'), ))
+    hetero_graph.nodes['n1'].data['label'] = th.randint(10, (hetero_graph.number_of_nodes('n1'), ))
+
+    hetero_graph.edges['r0'].data['feat'] = edge_feat['r0']
+    hetero_graph.edges['r1'].data['feat'] = edge_feat['r1']
+    hetero_graph.edges['r1'].data['label'] = th.randint(10, (hetero_graph.number_of_edges('r1'), ))
+
+    # set train/val/test masks for nodes and edges
+    if gen_mask:
+        target_ntype = ['n0', 'n1']
+        target_etype = [("n0", "r1", "n1"), ("n0", "r0", "n1")]
+
+        node_train_mask = generate_mask([0,1], data_size)
+        node_val_mask = generate_mask([2,3], data_size)
+        node_test_mask = generate_mask([4,5], data_size)
+
+        node_train_mask2 = generate_mask([i for i in range(data_size//2)], data_size)
+        node_val_mask2 = generate_mask([2,3], data_size)
+        node_test_mask2 = generate_mask([4,5], data_size)
+
+        edge_train_mask = generate_mask([0,1], 2 * data_size)
+        edge_val_mask = generate_mask([2,3], 2 * data_size)
+        edge_test_mask = generate_mask([4,5], 2 * data_size)
+
+        edge_train_mask2 = generate_mask([i for i in range(data_size//2)], data_size)
+        edge_val_mask2 = generate_mask([2,3], data_size)
+        edge_test_mask2 = generate_mask([4,5], data_size)
+
+        hetero_graph.nodes[target_ntype[0]].data['train_mask'] = node_train_mask
+        hetero_graph.nodes[target_ntype[0]].data['val_mask'] = node_val_mask
+        hetero_graph.nodes[target_ntype[0]].data['test_mask'] = node_test_mask
+
+        hetero_graph.nodes[target_ntype[1]].data['train_mask'] = node_train_mask2
+        hetero_graph.nodes[target_ntype[1]].data['val_mask'] = node_val_mask2
+        hetero_graph.nodes[target_ntype[1]].data['test_mask'] = node_test_mask2
+
+        hetero_graph.edges[target_etype[0]].data['train_mask'] = edge_train_mask
+        hetero_graph.edges[target_etype[0]].data['val_mask'] = edge_val_mask
+        hetero_graph.edges[target_etype[0]].data['test_mask'] = edge_test_mask
+
+        hetero_graph.edges[target_etype[1]].data['train_mask'] = edge_train_mask2
+        hetero_graph.edges[target_etype[1]].data['val_mask'] = edge_val_mask2
+        hetero_graph.edges[target_etype[1]].data['test_mask'] = edge_test_mask2
```

Comment:
why target_ntype graphs have target_etype?

Reply:
can we extend the existing generate_dummy_hetero_graph function?

Reply:
> why target_ntype graphs have target_etype?

I was following the original `generate_dummy_hetero_graph` function to make it close to the original function as much as possible

> can we extend the existing generate_dummy_hetero_graph function?

We cannot do this because in another unit-test [here](https://github.com/awslabs/graphstorm/blob/de978745ffad7557ca9f1effde836a8c060da27d/tests/unit-tests/test_dataloading.py#L279) we check whether another node type can fetch labels.